vault/identity_store_oidc_provider.go

```diff
@@ -1219,3 +1151,632 @@ func (i *IdentityStore) pathOIDCProviderExistenceCheck(ctx context.Context, req
 
 	return entry != nil, nil
 }
+
+func (i *IdentityStore) pathOIDCProviderDiscovery(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
```

Comment:
I moved some code around when resolving conflicts from a merge of main. The following methods have been reviewed and aren't being introduced in this PR:
- `pathOIDCProviderDiscovery`
- `pathOIDCReadProviderPublicKeys`
- `keyIDsReferencedByTargetClientIDs`